crates/bevy_color/src/color_ops.rs

```diff
@@ -80,6 +80,12 @@ pub trait Hue: Sized {
     }
 }
 
+/// Trait for returning a grayscale color of a provided lightness.
+pub trait Gray: Sized {
+    /// Returns a grey color with the provided lightness from (0.0 - 1.0). 0 is black, 1 is white.
+    fn gray(lightness: f32) -> Self;
```

Comment:
Since all the colours this is implemented for already have associated constants for `BLACK` and `WHITE`, and also implement `Mix`, perhaps those constants should be moved into this trait and a default implementation of the `gray` function could use `Mix`? For the simple spaces like Linear RGB I expect mix to perform identically.

I also think this would be a good sanity check for the `Mix` trait. I (and I imagine most people) would expect mixing black and white and some ration `t` to produce the same result as `gray(t)`. I don't think we would need to enforce that as an invariant, but I do think they should at least be close.

Reply:
I re-implemented the trait as having a default implementation from mix and srgb colors.

I didn't see many pre-built colors outside of srgb, so I used that as the source, then converted to the native format, then mixed. I did the mixing after conversion so different mix implementations would be used, which I hope covers lerping accross the color space, instead of just the numeric space.

Reply:
Sorry I wasn't very clear with my comment, I meant a default implementation for the _function_, not the trait:

```rust
pub trait Gray: Sized + Mix {
    const BLACK: Self;
    const WHITE: Self;

    fn gray(lightness: f32) -> Self {
        Self::BLACK.mix(&Self::WHITE, lightness)
    }
}
```

It is safe to assume every colour implements `Mix` (with the exception of `Color` itself, but a `gray` function for that type is ambiguous anyway), so it's ok to put in the trait bounds IMO.

This leaves open the possibility of overriding the specific `gray` implementation for certain colour spaces as desired.

Finally, it would ensure `WHITE` and `BLACK` constants are available in a standard way for all colours (which would already be the case since they would be axiomatically `gray(1.)` and `gray(0.)`)

Reply:
That is a much better approach! And I think that resolves needing to import Grey to use the function.

I'll see about creating implementations for the different color spaces.

Reply:
It does not resolve needing to import Grey, but it is cleaner to implement.

Reply:
I pulled the constants for white and black from `crates\bevy_color\src\test_colors.rs`, that file leads me towards wanting to generate common color values for every color space (pretty much make the test data actual data). But, that's a later PR!